src/main/java/uk/gov/justice/digital/hmpps/whereabouts/config/WebClientConfiguration.java

```diff
@@ -0,0 +1,154 @@
+                    .build();
+            return next.exchange(filtered);
+        };
+    }
+
+    @Bean
+    @RequestScope
+    public WebClient elite2WebClient(final ClientRegistrationRepository clientRegistrationRepository,
+                                     final OAuth2AuthorizedClientRepository authorizedClientRepository,
+                                     final WebClient.Builder builder) {
+        return getOAuthWebClient(authorizedClientManager(clientRegistrationRepository, authorizedClientRepository), builder, elite2ApiRootUri);
+    }
+
+    @Bean
+    @RequestScope
+    public WebClient caseNoteWebClient(final ClientRegistrationRepository clientRegistrationRepository,
+                                       final OAuth2AuthorizedClientRepository authorizedClientRepository,
+                                       final WebClient.Builder builder) {
+
+        return getOAuthWebClient(authorizedClientManager(clientRegistrationRepository, authorizedClientRepository), builder, caseNotesRootUri);
+    }
+
+    @Bean
+    public OAuth2AuthorizedClientManager authorizedClientManagerAppScope(final ClientRegistrationRepository clientRegistrationRepository,
+                                                                         final OAuth2AuthorizedClientService oAuth2AuthorizedClientService) {
+
+        final var authorizedClientProvider = OAuth2AuthorizedClientProviderBuilder.builder().clientCredentials().build();
+        final var authorizedClientManager = new AuthorizedClientServiceOAuth2AuthorizedClientManager(clientRegistrationRepository, oAuth2AuthorizedClientService);
+
+        authorizedClientManager.setAuthorizedClientProvider(authorizedClientProvider);
+        return authorizedClientManager;
+    }
+
+    @Bean
+    public WebClient elite2WebClientAppScope(@Qualifier(value = "authorizedClientManagerAppScope") final OAuth2AuthorizedClientManager authorizedClientManager, final WebClient.Builder builder) {
+         return getOAuthWebClient(authorizedClientManager, builder, elite2ApiRootUri);
+    }
+
+    @Bean
+    public WebClient caseNoteWebClientAppScope(@Qualifier(value = "authorizedClientManagerAppScope") final OAuth2AuthorizedClientManager authorizedClientManager, final WebClient.Builder builder) {
+        return getOAuthWebClient(authorizedClientManager, builder, caseNotesRootUri);
+    }
+
+    private WebClient getOAuthWebClient(final OAuth2AuthorizedClientManager authorizedClientManager, final WebClient.Builder builder, final String rootUri) {
+        final var oauth2Client = new ServletOAuth2AuthorizedClientExchangeFilterFunction(authorizedClientManager);
+        oauth2Client.setDefaultClientRegistrationId("elite2-api");
+
+        final var exchangeStrategies =
+                ExchangeStrategies.builder()
+                        .codecs(configurer -> configurer.defaultCodecs().maxInMemorySize(-1))
+                        .build();
```

Comment:
It later transpired that this only affects one of the many decoders that are used, and we had to set a Spring property to set the max size (keyorker-api PR #240).